machine.go

```diff
@@ -579,11 +579,16 @@ func (m *Machine) setupLogging(ctx context.Context) error {
 func captureFifoToFile(logger *log.Entry, fifoPath string, fifo io.Writer) error {
 	// create the fifo pipe which will be used
 	// to write its contents to a file.
-	fifoPipe, err := os.OpenFile(fifoPath, os.O_RDONLY, 0600)
+	fd, err := syscall.Open(fifoPath, syscall.O_RDONLY|syscall.O_NONBLOCK, 0600)
 	if err != nil {
 		return fmt.Errorf("Failed to open fifo path at %q: %v", fifoPath, err)
 	}
 
+	fifoPipe := os.NewFile(uintptr(fd), fifoPath)
+	if fifoPipe == nil {
+		return fmt.Errorf("Invalid file descriptor: %v", err)
```

Comment:
`os.NewFile()` doesn't set `err`. So we shouldn't include err in the error message since it would be always nil.

Reply:
Fixed!